### PR TITLE
ug: offline-update: Add info on bundle signing

### PR DESCRIPTION
**Pull request by mike-sul**
Add information about signing of the offline update bundle. Specifically, how to do it, and what errors to expect if the signature is invalid.

# PR Template and Checklist

Please complete as much as possible to speed up the reviewing process.

Readiness and adding reviewers as appropriate is required.

All PRs should be reviewed by a technical writer/documentation team and a peer.
If effecting customers—which is a majority of content changes—a member of Customer Success must also review.

## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

* [X] Run spelling and grammar check, preferably with linter.
* [X] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for b406c3b7e953a2cad7c89b70bc5d5afce4bc20ed are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2648/docs/artifacts/html/index.html

**Review by detsch — APPROVED** [review]
LGTM

**Review by angolini — APPROVED** [review]
I commented a lot about terms and standardization, but I know some of them are already on the page, some are not, and only greping the source code may not be enough to determine whether a term exists.

Other than that, I don't have any comments. The terms confused me a bit, but I think it's because reviewing a text differs from learning from it. So, feel free to ignore my comments.

File: source/user-guide/offline-update/offline-update.rst [hunk 1]
```diff
@@ -64,6 +64,24 @@ Use the command ``fioctl targets offline-update <target-name> <dst> <--tag <tag>
 .. note::
     In order to download all artifacts, ``fioctl`` requires token with scopes: ``targets:read``, ``ci:read``.
 
+Signing the Offline Update Bundle
+---------------------------------
+It is essential to sign the bundle using one or more TUF targets role keys.
+This ensures the authenticity of an offline update bundle during the update process on a device.
+
+If the bundle contains CI targets, it is signed by the OTA Lite service using the online TUF targets role key.
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply source/user-guide/offline-update/offline-update.rst:72]: > "online TUF targets role key"
> 
> is it role or root?

It is a role.

File: source/user-guide/offline-update/offline-update.rst [hunk 2]
```diff
@@ -64,6 +64,24 @@ Use the command ``fioctl targets offline-update <target-name> <dst> <--tag <tag>
...
+Users do not need to take any action in this scenario.
+
+If the bundle contains production or wave targets, it should be signed using one or more TUF targets offline keys.
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply source/user-guide/offline-update/offline-update.rst:75]: > "CI Target" is, but if it happens to be a Target built by CI, it's not a term.

It is something that is built by CI and can be deployed to devices. 
"wave target" - the target to which a given wave refer to.
"production target" - the target that can be deployed only to production devices.

CI build produces "CI target", then a user can make it "wave target" through the "fioctl waves init" command. Once a wave is complete all its targets become "production targets".

"TUF targets" or "TUF targets role" refers to TUF metadata as is defined in https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#targets-role--targets.

**Review by vkhoroz — APPROVED** [review]
LGTM

**Review by MrCry0 — APPROVED** [review]

**Comment by doanac** [comment]
Docs for f460a3e2cf59df8465d10b92e8f471a260c36f3f are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2650/docs/artifacts/html/index.html

File: source/user-guide/offline-update/offline-update.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (angolini) [reply source/user-guide/offline-update/offline-update.rst]: those pages are really complex and I'm not sure they belong to user guide, but at least we are using the same terms, only need to make sure there is a link to the external glossary.

regarding the CI target. Everything is a CI target, until it becomes a wave target or a production target.

Anyway, as I said, all those pages (not only the one in this PR) should be reviewed in the future to try to clean/organize them better. They are starting to become too dense for a user guide, and I think they deserve to go to reference manual. Anyway, it's a future work.

**Pull request merged**